File: OPUS-tauri-app/src/types/layout.ts
export type TerminalLayout = {
  version: number;
  root: LayoutNode;
  focused_pane_id?: string;
};

export type LayoutNode = PaneNode | SplitNode;

export type PaneNode = {
  kind: 'pane';
  id: string;
  tabs: LayoutTab[];
  active_tab_id?: string;
};

export type SplitNode = {
  kind: 'split';
  id: string;
  direction: 'row' | 'column';
  ratio: number;
  first: LayoutNode;
  second: LayoutNode;
};

export type LayoutTab = {
  id: string;
  terminal_id: string;
  title: string;
  kind: 'agent' | 'terminal' | 'diff';
  diff_repo?: string;
  diff_repo_path?: string;
  diff_file_path?: string;
  diff_prev_path?: string;
  diff_status?: string;
};

File: OPUS-tauri-app/src-tauri/src/types/layout.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct TerminalLayout {
    pub version: u32,
    pub root: LayoutNode,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub focused_pane_id: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(tag = "kind")]
pub enum LayoutNode {
    #[serde(rename = "pane")]
    Pane {
        id: String,
        tabs: Vec<LayoutTab>,
        #[serde(skip_serializing_if = "Option::is_none")]
        active_tab_id: Option<String>,
    },
    #[serde(rename = "split")]
    Split {
        id: String,
        direction: String,
        ratio: f64,
        first: Box<LayoutNode>,
        second: Box<LayoutNode>,
    },
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct LayoutTab {
    pub id: String,
    pub terminal_id: String,
    pub title: String,
    pub kind: String,
}
